client.go

```diff
@@ -426,3 +433,20 @@ func (c *Client) createComponents(config *Config) {
 	c.proxyManager = newProxyManager(proxyManagerServiceBundle)
 	c.invocationHandler = invocationHandler
 }
+
+func addrProviderTranslator(ctx context.Context, config *cluster.Config, logger ilogger.Logger) (pr icluster.AddressProvider, tr icluster.AddressTranslator, err error) {
+	if config.HazelcastCloudConfig.Enabled {
+		dc := cloud.NewDiscoveryClient(&config.HazelcastCloudConfig, logger)
+		nodes, err := dc.DiscoverNodes(ctx)
+		if err != nil {
+			return nil, nil, err
+		}
+		if pr, err = cloud.NewAddressProvider(nodes); err != nil {
+			return nil, nil, err
+		}
+		tr = cloud.NewAddressTranslator(dc, nodes)
+	} else {
+		pr = icluster.NewDefaultAddressProvider(config)
```

Comment:
Where do we create address translators now, when the cloud discovery is not enabled?

Reply:
It seems I've botched that part of the code during the reviews. I've fixed that.

Reply:
At: https://github.com/hazelcast/hazelcast-go-client/pull/573/commits/ab6d150c89ab7b36e5b312adc9cb08a5200a3531